### ISSUE TITLE
Rules inside `template` of Vue components

### ISSUE THREAD

**Issue by Ericlm** [eslint-stylistic/eslint-stylistic#419]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

In my Vue project, some stylistic rules don't seem to apply when inside the template of a Vue component.
For example, when inside the script, the rule is reported : 
```ts
hi.value&&hello.value // Operator '&&' must be spaced.
```
But inside the template, there is no report:
```vue
<HelloWorld msg="You did it!" v-if="hi&&hello" /> // No issue detected
```
More precisely, there is no report about the JS inside the template; the other rules like `no-trailing-spaces` works perfectly fine everywhere.

I guess that the issue lies not directly in Stylistic but in the parsing of the template maybe?

### Reproduction

https://github.com/Ericlm/vue-stylistic/blob/main/src/App.vue

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by Ericlm** [comment]
In fact it's made by the eslint-plugin-vue which extends stylistic rules (see [extension rules](https://eslint.vuejs.org/rules/#extension-rules)).
Also: https://github.com/vuejs/eslint-plugin-vue/issues/2480
